### ISSUE TITLE
Don't treat destructuring fields as logic vars

### ISSUE THREAD

**Issue by FiV0** [xtdb/crux#2881]
There are currently a couple of issues regarding destructuring that seem to treat the field to be destructered as logic var.
```clj
(deftest allow-for-destructering-in-with
  (t/is (= nil
           (xt/q tu/*node*
                 '(-> (table [{:x {:first_name "Ivan" :last_name "Ivanov"}}] [x])
                      (with {:first-name (. x first-name)}))))))
```

**Comment by jarohen** [comment]
closing unless this also affects XTQL? @FiV0 could you re-open if so?

**Comment by FiV0** [comment]
Updated the card to XTQL

**Comment by jarohen** [comment]
Ah, gotcha. I wonder whether this means we need to have explicit support for a get-field expr in the AST, rather than treating it as a call?

**Comment by jarohen** [comment]
@mbutlerw this fixed in #2955 ?
